src/autotype/xcb/AutoTypeXCB.cpp

```diff
@@ -802,8 +814,11 @@ void AutoTypePlatformX11::SendKeyPressedEvent(KeySym keysym)
 
     /* press and release key */
     event.keycode = keycode;
-    SendEvent(&event, KeyPress);
-    SendEvent(&event, KeyRelease);
+    if (isKeyDown) {
```

Comment:
With this change, modifiers will be pressed and released for each key event.
Instead of modifing the code here, you can use `SendEvent` and `SendModifier` directly in execClearField.